portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -23,13 +23,28 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
 /** Servlet responsible for deleting comments. */
 @WebServlet("/delete-comment")
 public class DeleteServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Double check to see if user is logged in
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+    // Double check if comment to be deleted is authored by current user
+    if (request.getParameter("isAuthor").equals("false")){
```

Comment:
you might want to read from datastore for this check to make it safer, in case there is a request not generated by your ui sent to your server deleting other's comments